Posix Daemon mode can't be built

**Describe the bug**
Trying to build OpenThread Posix Daemon results in a config check error.

**To Reproduce**
1. Clone openthread from master
1. Run `make -f src/posix/Makefile-posix DAEMON=1`
1. Build errs out with many of these errors:

In file included from /openthread/src/posix/../../src/core/openthread-core-config.h:89,
                 from /openthread/src/posix/../../third_party/mbedtls/mbedtls-config.h:32,
                 from /openthread/src/posix/../../third_party/mbedtls/repo/library/cmac.c:46:
/openthread/src/posix/../../src/core/config/openthread-core-config-check.h:38:2: error: #error "OPENTHREAD_CONFIG_PLATFORM_UDP_ENABLE and OPENTHREAD_CONFIG_UDP_FORWARD_ENABLE must not both be set."
   38 | #error "OPENTHREAD_CONFIG_PLATFORM_UDP_ENABLE and OPENTHREAD_CONFIG_UDP_FORWARD_ENABLE must not both be set."


**Expected behavior**
ot-daemon is built.

**Additional context**
Following the instructions here: https://github.com/openthread/openthread/blob/master/src/posix/README.md#daemon-mode-support

--- a/src/posix/Makefile-posix
+++ b/src/posix/Makefile-posix
@@ -64,7 +64,9 @@ READLINE                             ?= readline
 REFERENCE_DEVICE                     ?= 1
 SERVICE                              ?= 1
 SNTP_CLIENT                          ?= 1
+ifneq ($(DAEMON),1)
 UDP_FORWARD                          ?= 1
+endif
 
 COMMONCFLAGS                         := \
     -g                                  \

